rhp/v4/validation.go

```diff
@@ -149,84 +153,83 @@ func (req *RPCFormContractRequest) Validate(pk types.PublicKey, tip types.ChainI
+		return fmt.Errorf("renewal proof height must be greater than %v", minProofHeight)
+	case expirationHeight > maxExpirationHeight:
+		return fmt.Errorf("renewal expiration height %v exceeds max expiration height %v", expirationHeight, maxExpirationHeight)
 	}
 
 	// validate the contract fields
 	hp := req.Prices
-	expirationHeight := req.Renewal.ProofHeight + ProofWindow
 	duration := expirationHeight - hp.TipHeight
 	// calculate the minimum allowance required for the contract based on the
 	// host's locked collateral and the contract duration
 	minRenterAllowance := MinRenterAllowance(hp, req.Renewal.Collateral)
 	// collateral is risked for the entire contract duration
-	riskedCollateral := req.Prices.Collateral.Mul64(existingSize).Mul64(expirationHeight - req.Prices.TipHeight)
+	riskedCollateral := req.Prices.Collateral.Mul64(existing.Filesize).Mul64(duration)
 	// renewals add collateral on top of the required risked collateral
 	totalCollateral := req.Renewal.Collateral.Add(riskedCollateral)
 
 	switch {
-	case expirationHeight <= tip.Height: // must be validated against tip instead of prices
-		return errors.New("contract expiration height is in the past")
 	case req.Renewal.Allowance.IsZero():
 		return errors.New("allowance must be greater than zero")
 	case totalCollateral.Cmp(maxCollateral) > 0:
 		return fmt.Errorf("required collateral %v exceeds max collateral %v", totalCollateral, maxCollateral)
-	case duration > maxDuration:
-		return fmt.Errorf("contract duration %v exceeds max duration %v", duration, maxDuration)
 	case req.Renewal.Allowance.Cmp(minRenterAllowance) < 0:
 		return fmt.Errorf("allowance %v is less than minimum allowance %v", req.Renewal.Allowance, minRenterAllowance)
 	default:
 		return nil
 	}
 }
 
-// Validate validates a refresh contract request. Prices are not validated
-// pk - the public key of the host
-// existingCollateral - the existing, unallocated collateral of the contract
-// existingTotalCollateral - the existing total (allocated+unallocated) collateral of the contract
-// existingAllowance - the existing, remaining allowance of the contract
-func (req *RPCRefreshContractRequest) Validate(pk types.PublicKey, existingCollateral, existingTotalCollateral, existingAllowance types.Currency, expirationHeight uint64, maxCollateral types.Currency) error {
+// Validate validates a refresh contract request against the existing contract.
+func (req *RPCRefreshContractRequest) Validate(pk types.PublicKey, tip types.ChainIndex, existing types.V2FileContract, maxCollateral types.Currency) error {
 	if err := req.Prices.Validate(pk); err != nil {
 		return fmt.Errorf("prices are invalid: %w", err)
 	}
 
 	// validate the request fields
+	minProofHeight := minProofHeight(tip, req.Prices)
 	switch {
 	case req.MinerFee.IsZero():
 		return errors.New("miner fee must be greater than 0")
 	case req.Basis == (types.ChainIndex{}):
 		return errors.New("basis must be set")
+	case existing.ProofHeight <= minProofHeight:
```

Comment:
shouldn't this be strictly less, considering the docs on `minProofHeight` indicate it's the minimum height necessary for the host to accept the contract